mlos_bench/mlos_bench/optimizers/mlos_core_optimizer.py

```diff
@@ -202,7 +206,18 @@ def get_best_observation(self) -> Union[Tuple[Dict[str, float], TunableGroups],
         (df_config, df_score, _df_context) = self._opt.get_best_observations()
         if len(df_config) == 0:
             return (None, None)
+
         params = configspace_data_to_tunable_values(df_config.iloc[0].to_dict())
         scores = self._adjust_signs_df(df_score).iloc[0].to_dict()
+
+        # Check for NaN values in all optimization targets and replace with 0
+        for target in self._opt_targets:
+            if target in scores:
+                if pd.isna(scores[target]):
+                    _LOG.warning(f"'{target}' is NaN in the best observation. Setting it to 0.")
+                    scores[target] = 0
```

Comment:
I think this should be optional behavior.  It might be better to mark the Trial as FAILED instead and ask the user to check their scripts.

Reply:
See also #523

Reply:
I agree setting nan to 0 is not ideal but I desperately wanted long running experiments

Reply:
I think it is totally fine to report NaNs here - say, for scores that were not the primary optimization target or at all absent in a particular trial (e.g., because of the experiments' merge). Not to mention that 0 may not be the right value to impute, depending on your optimization direction. I am afraid that @yshady is bending MLOS to make the UI work smoothly and I think it should be the other way around :)

Reply:
@motus no the UI is separate from MLOS experiments. All it does is run the commands. Yes these fixes were to just make MLOS run for long periods of time and they are shortcuts and not ideal. @eujing can confirm as well, I am bending MLOS to just work :) not just for the UI but for the sake of collecting data

Reply:
@motus also optimizer does not like nan values from my experience

Reply:
I will try to impute the correct value tomorrow